### PR TITLE
Remove Turbolinks

### PR DESCRIPTION
**Pull request by mheap**
## Description

Turbolinks has been causing issues where JS isn't executed as it's not
a real page refresh. This means we keep adding workarounds for third
party services

This commit removes Turbolinks, and as far as I can tell it doesn't have
a negative performance impact.

## Deploy Notes

Check things thoroughly with this deploy as it's quite a big change

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -12,14 +12,14 @@
...
 //= require jquery_ujs
-//= require turbolinks
 //= require jquery-scrolltofixed
 //= require volta/volta.core.js
 //= require volta/components/volta.accordion.js
 //= require volta/components/volta.tab.js
 //= require volta/components/volta.tooltip.js
 //= require volta/components/volta.modal.js
 //= require volta/components/volta.dropdown.js
+//= require volta/components/volta.menu.js
```
Comment (benhayehudi) [inline app/assets/javascripts/application.js]: Are we also including the Volta menu scripts as part of this PR? Is that a separate issue?

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -67,15 +56,7 @@ let refresh = () => {
...
-  Volta.init(['accordion', 'tooltip', 'tab', 'modal', 'dropdown'])
-
-  // Fix for Turbolinks scrolling to in-page anchor when navigating to a new page
-  if(window.location.hash){
-    const tag = document.getElementById(window.location.hash.slice(1))
-    if(tag){
-      tag.scrollIntoView(true);
-    }
-  }
+  Volta.init(['accordion', 'tooltip', 'tab', 'modal', 'dropdown', 'menu'])
```
Comment (benhayehudi) [inline app/javascript/packs/application.js]: Same question as above.

File: app/pipelines/markdown_pipeline.rb [hunk 3]
```diff
@@ -35,8 +35,7 @@ def initialize(options = {})
       BreakFilter,
       UnfreezeFilter,
       IconFilter,
-      ExternalLinkFilter,
-      TutorialLinkFilter
```
Comment (benhayehudi) [inline app/pipelines/markdown_pipeline.rb]: How is this connected to the removal of Turbolinks?

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (mheap) [reply app/assets/javascripts/application.js]: This is intended. We used to have a `Navigation.js` file that contained a copy of the Volta menu code with some Turbolinks enhancements. Now that we've removed Turbolinks we can use the standard Volta menu code

File: app/pipelines/markdown_pipeline.rb (hunk 3 above)
Comment (mheap) [reply app/pipelines/markdown_pipeline.rb]: The `TutorialLinkFilter` detected if any link starts with `/tutorial` and added an attribute to disable turbolinks for that link, which is no longer needed

File: vendor/assets/javascripts/volta/components/volta.menu.js [hunk 4]
```diff
@@ -0,0 +1,377 @@
+/**
```
Comment (fabianrbz) [inline vendor/assets/javascripts/volta/components/volta.menu.js]: @mheap can we use [this one](https://github.com/Nexmo/nexmo-developer/blob/master/vendor/assets/javascripts/volta/components/side-navigation/volta.menu.js) instead?

**Pull request merged**